docs/design/fabric-v2+/fpc-registration.puml

```diff
@@ -7,266 +7,198 @@
+  activate ERCC2
+
+  ERCC1 -> ERCC1 : check Credentials.Evidence
+  ERCC1 -> ERCC1 : check Credentials.Attested_Data.CC_Parameters
+
+  group check consistency of Credentials (TBD)
+    note over ERCC1 #FFaa00
+        It is crucial for ERCC to perform queries over (or have access to) the chaincode
+        definition. Querying (or retrieving) the chaincode definition through TLCC is the
+        most convenient way to implement this with least impact on core Fabric code.
+        Future tighter integration in Fabric could also change that to more direct ways
+        to feed it to ERCC.
     end note
-
-    ERCC1 -> ERCC1 : Retrieve ECC definition for <CC_NAME, CC_VER>
-
-    ERCC1 -> ERCC1 : Verify SIG_Org_1
-
-    ERCC1 -> ERCC1 : check consistency of quote (report-body)
-    note left
-      ECC_MR (mrenclave of ecc) is defined in ECC definition;
-      TLCC_MR (mrenclave of tlcc) is defined in ERCC definition
-      CH_Id is read from ledger or defined in ERCC definition
+    create "TLCC Peer1_Org1" as TLCC1 order 45
+    ERCC1 -> TLCC1  ++: approve consistency of Credentials
+    TLCC1 -> TLCC1    : check Credentials against chaincode definition
+    note over TLCC1 #FFaa00
+        This check ensures that:
+        1) the chaincode version (i.e., the MREnclave) matches the version of the
+            approved chaincode.
+        2) the chaincode is registered
+        Note that (1) is security-critical and saves the client from verifying the
+        chaincode version/MREnclave.
+        Note that (2) merely prevents spurious chaincode enclaves from being registered.
     end note
+    return result
+  end group
 
+  group deployment validation (unclear, TBD)
     ERCC1 -> ERCC1 : validate FPC deployment (restriction) policy
     note left #FFaa00
       related risk management (v2.0 feature TBD)
     end note
-    ERCC1 -> ERCC1 : putState
-    note left #FFaa00
-      key: CC_NAME + (CC_VER???) + ECC_ID
-      Note that this key construction allows efficient lookup (details TBD)
+  end group
 
-      value: <CC_NAME, CC_VER, MSP_ID_ORG_1, SIG_Org_1, ECC_S_PK, ECC_E_PK, ATTESTATION>
-    end note
-
-  end
+  ERCC1 -> ERCC1 : store Credentials on the ledger
+  note over ERCC1 #FFaa00
+  E.g., putState(key=Credentials.CC_Id, value=Credentials)
+  end note
 
-  return register endorsement
+  return register endorsement /' E_Manager -> ERCC2 '/
 
-  Admin -> ERCC2 ++: register (CC_NAME, CC_VER, MSP_ID_ORG_1, SIG_Org_1, ECC_S_PK, ECC_E_PK, ATTESTATION)
-  ref over ERCC2
-    verify and store
-  end ref
-  return register endorsement
+  return register endorsement /' E_Manager -> ERCC1 '/
 
-  Admin -> ERCC3 ++: register (CC_NAME, CC_VER, MSP_ID_ORG_1, SIG_Org_1, ECC_S_PK, ECC_E_PK, ATTESTATION)
-  ref over ERCC3
-    verify and store
+  ref over E_Manager
+    submit transaction
   end ref
-  return register endorsement
-
-  Admin -> Admin : create register transaction
 
-  Admin -> Orderer : submit register transaction
+  return result
```

Comment:
skipping this will not complete the enclave registration.

Reply:
aha ... now I see in line 195 the `submit transaction` action .... this makes me wondering what the manager actually is ... part of the peer or part of a client ... see related question above also.

1. If the manager is part of the peer, it cannot submit transactions, in particular, it cannot send transaction proposals to other endorsers, collect endorsements and submit a tx. It just has no client credentials for that. 
1. If the manager is part of a client (admin). it could indeed perform the actions just mentioned using the admin credentials ...